Mathlib/CategoryTheory/Limits/Shapes/Preorder/HasIterationOfShape.lean

```diff
@@ -51,4 +60,42 @@ instance : HasIterationOfShape J (K ⥤ C) where
     have := hasColimitsOfShape_of_isSuccLimit C j hj
     infer_instance
 
+variable {J} [SuccOrder J] [WellFoundedLT J]
+
+lemma hasColimitsOfShape_of_initialSeg
+    {α : Type*} [LinearOrder α] (f : α ≤i J) [Nonempty α] :
+    HasColimitsOfShape α C := by
+  by_cases hf : Function.Surjective f
+  · exact hasColimitsOfShape_of_equivalence
+      (OrderIso.ofRelIsoLT (RelIso.ofSurjective f.toRelEmbedding hf)).equivalence.symm
+  · let s := f.toPrincipalSeg hf
+    obtain ⟨i, hi₀⟩ : ∃ i, i = s.top := ⟨_, rfl⟩
+    induction i using SuccOrder.limitRecOn with
+    | hm i hi =>
+        subst hi₀
```

Comment:
The style guide isn't explicit about this, but the usual style in mathlib is to indent this as
```suggestion
      subst hi₀
```